### PR TITLE
Task3 feat(webpack) : use webpack in dev/prod modes

### PR DESCRIPTION
**Pull request by pblinkov**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"singleQuote": true,
+	"useTabs": true
```
Comment (boccob) [inline .prettierrc:3]: I'd prefer spaces because it's easier to review (GitHub shows quite wide tabs).

File: package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		"clean-webpack-plugin": "^1.0.0",
+		"css-loader": "^1.0.1",
+		"eslint": "^5.9.0",
+		"eslint-config-prettier": "^3.3.0",
+		"extract-loader": "^3.1.0",
+		"extract-text-webpack-plugin": "^3.0.2",
+		"file-loader": "^2.0.0",
+		"html-loader": "^0.5.5",
+		"html-webpack-plugin": "^3.2.0",
+		"husky": "^1.2.0",
```
Comment (boccob) [inline package.json:35]: You need to configure it, otherwise it's useless.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		"less": "^3.8.1",
+		"less-loader": "^4.1.0",
+		"mini-css-extract-plugin": "^0.4.5",
+		"prettier": "^1.15.2",
```
Comment (boccob) [inline package.json:39]: It'd be better to create npm script that will check / fix code formatting.

File: src/app.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import '@babel/polyfill';
+import Element from './element/element.component';
+import Wrapper from './wrapper/wrapper.component';
+import FetchService from './services/fetch.component';
+import Article from './article/article.component';
+import { helper } from './helper';
+import './less/main.less';
+
+const fetchService = new FetchService;
+let articleArr = [];
```
Comment (boccob) [inline src/app.js]: Do you use it?

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+const Button = new Element('button');
+
+Button.elem.type = 'button';
+Button.elem.classList.add('button');
+Button.elem.innerHTML = 'Show articles';
+
+
+HeaderWrapper.append(Button.elem);
+
+Header.append(HeaderWrapper.elem);
```
Comment (boccob) [inline src/app.js:29]: Final structure of elements tree is not very obvious now. It's not critical but try to think how to make it better.

File: src/app.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+Main.append(MainWrapper.elem);
+Footer.append(FooterWrapper.elem);
+
+App
+	.append(Header.elem)
+	.append(Main.elem)
+	.append(Footer.elem)
+	.makeFragment()
```
Comment (boccob) [inline src/app.js:37]: Is there any case when you make a fragment but don't want to append it?

File: src/app.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+	.appendToElement();
+
+
+
+
+
+
+helper.on(helper.find('button'), 'click', e => {
+	fetchService.request()
+		.then(function(response) {
```
Comment (boccob) [inline src/app.js]: Why not arrow function?

File: src/app.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+			return response.json();
+		})
+		.then(responce => {
```
Comment (boccob) [inline src/app.js]: *response

File: src/article/article.component.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import Element from '../element/element.component';
+import './article.style.less';
+
+export default class Article extends Element{
```
Comment (boccob) [inline src/article/article.component.js:4]: Well, I'm pretty sure that you don't use `prettier`.

File: webpack.common.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+		path: path.resolve(__dirname, 'dist')
+	},
+	module: {
+		rules: [
+			{
+				test: /\.less$/,
+				use: [
+					MiniCssExtractPlugin.loader,
+					"css-loader",
+					"less-loader"
```
Comment (boccob) [inline webpack.common.js:28]: Would be good to add autoprefixer.

File: webpack.common.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+	},
+	plugins: [
+		new HtmlWebpackPlugin({title: 'Production'}),
+		new MiniCssExtractPlugin({
+			filename: "[name].css",
+			chunkFilename: "[id].css"
+		})
+	],
+	output: {
+		filename: '[name].bundle.js',
```
Comment (boccob) [inline webpack.common.js:17]: Usually it's a good idea to add hash for production build. So, it'll be pretty straightforward to enable caching for all static files.

**Review by boccob — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (pblinkov) [reply package.json:39]: I use WS with ESLint plugins. This plugin is on now.

**Pull request merged**